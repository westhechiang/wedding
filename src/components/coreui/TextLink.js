import styled from 'styled-components';
import { theme } from '../../ui';

const sizes = {
  large: theme.font.large,
  base: theme.font.base,
  small: theme.font.small,
  xsmall: theme.font.xsmall,
  xxsmall: theme.font.xxsmall,
  xxxsmall: theme.font.xxxsmall,
};

const weights = {
  light: theme.font.weight.light,
  undefined: theme.font.weight.medium,
};

const hoverEffects = {
  boldFont: `font-weight: ${theme.font.weight.semiBold};`,
};

const TextLink = styled.a`
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: ${props => theme.color[props.color]};
  font-size: ${props => sizes[props.size]};
  font-family: ${theme.font.family};
  font-weight: ${props => weights[props.weight]};
  line-height: ${props => sizes[props.size]};
  text-indent: ${props => props.textIndent};
  text-align: ${props => (props.textAlign || 'start')};
  text-transform: ${props => (props.uppercase ? 'uppercase' : 'none')};
  margin: ${props => (props.margin || 0)};
  position: ${props => (props.position || 'relative')};
  top: ${props => (props.top || 'auto')};
  padding: ${props => (props.padding || '0')};
  opacity: ${props => props.opacity || '1'};
  white-space: ${props => props.whiteSpace || 'normal'};
  overflow: ${props => props.overflow || 'visible'};
  text-overflow: ${props => props.textOverflow || 'clip'};

  &:hover {
    cursor: pointer;
    ${props => hoverEffects[props.hover]}
  }
`;

export default TextLink;
