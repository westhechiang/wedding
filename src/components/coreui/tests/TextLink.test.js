import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextLink from '../../../components/coreui/TextLink';

describe('<TextLink />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<TextLink />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should match its "large" size snapshot', () => {
    const wrapper = shallow(<TextLink size="large" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "base" size snapshot', () => {
    const wrapper = shallow(<TextLink size="base" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "small" size snapshot', () => {
    const wrapper = shallow(<TextLink size="small" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xsmall" size snapshot', () => {
    const wrapper = shallow(<TextLink size="xsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xxsmall" size snapshot', () => {
    const wrapper = shallow(<TextLink size="xxsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "xxxsmall" size snapshot', () => {
    const wrapper = shallow(<TextLink size="xxxsmall" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "light" weight snapshot', () => {
    const wrapper = shallow(<TextLink weight="light" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "semiBold" weight snapshot', () => {
    const wrapper = shallow(<TextLink weight="semiBold" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "boldFont" hover effect snapshot', () => {
    const wrapper = shallow(<TextLink hover="boldFont" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
