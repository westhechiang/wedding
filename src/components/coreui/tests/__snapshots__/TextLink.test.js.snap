// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> should match its "base" size snapshot 1`] = `
<a
  className="sc-bdVaJa jngQPP"
  size="base"
/>
`;

exports[`<TextLink /> should match its "boldFont" hover effect snapshot 1`] = `
<a
  className="sc-bdVaJa lgNApY"
/>
`;

exports[`<TextLink /> should match its "large" size snapshot 1`] = `
<a
  className="sc-bdVaJa ceLlWj"
  size="large"
/>
`;

exports[`<TextLink /> should match its "light" weight snapshot 1`] = `
<a
  className="sc-bdVaJa dbEpgK"
/>
`;

exports[`<TextLink /> should match its "semiBold" weight snapshot 1`] = `
<a
  className="sc-bdVaJa gyFwkz"
/>
`;

exports[`<TextLink /> should match its "small" size snapshot 1`] = `
<a
  className="sc-bdVaJa ezttOt"
  size="small"
/>
`;

exports[`<TextLink /> should match its "xsmall" size snapshot 1`] = `
<a
  className="sc-bdVaJa jvNczD"
  size="xsmall"
/>
`;

exports[`<TextLink /> should match its "xxsmall" size snapshot 1`] = `
<a
  className="sc-bdVaJa fCSmvd"
  size="xxsmall"
/>
`;

exports[`<TextLink /> should match its "xxxsmall" size snapshot 1`] = `
<a
  className="sc-bdVaJa eypxMo"
  size="xxxsmall"
/>
`;

exports[`<TextLink /> should match its empty snapshot 1`] = `
<a
  className="sc-bdVaJa hQXzYs"
/>
`;
